### PR TITLE
Add pre-commit hook for running addlicense

### PR DESCRIPTION
**Pull request by rickeylev**
Integrate the addlicense.sh script with the pre-commit config so that it automatically runs when git committing.

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- id: addlicense
```
Comment (alexeagle) [inline .pre-commit-hooks.yaml:1]: what is the function of this added file? what calls it?

**Comment by alexeagle** [comment]
The standard way to do this is expose a pre-commit hook from the addlicense repo, so that everyone can add it, and you don't need a shell script or a new `.pre-commit-hooks.yaml` here

**Comment by rickeylev** [comment]
Thanks for the pointer. I sent https://github.com/google/addlicense/pull/143. Once accepted (I don't see why it wouldn't be...), we'll have to wait for them to do a release with a stable tag to use in the configs. I have another PR ready for when that happens.

In the meantime, I found the docs that tell how to correctly setup a local hook and updated the code to do that. PTAL.

**Comment by f0rmiga** [comment]
It bothers me that pre-commit doesn't use Bazel for the binaries. The best way to deal with this is:

- Pull the binaries using http_archives.
- Create an alias with selects to resolve the platform.
- Create a shell wrapper that cd into `BUILD_WORKSPACE_DIRECTORY` and execute the binary:
```shell
#!/usr/bin/env bash

set -o errexit -o nounset

cd "${BUILD_WORKSPACE_DIRECTORY}"

exec "${ADDLICENSE_BINARY}" "$@"
```
- Add the `entry` attribute to the pre-commit hook with `entry: "bazel run //<pre-commit wrappers package>:addlicense --"`.

**Comment by chrislovecnm** [comment]
Can we get this into a test?  CI should fail on this.

**Comment by chrislovecnm** [comment]
@rickeylev ping

**Pull request closed without merging**